monai/data/iterable_dataset.py

```diff
@@ -80,36 +99,31 @@ def __init__(self, data, transform=None, buffer_size: int = 512, seed: int = 0)
         self.seed = seed
         self._idx = 0
 
+    def randomized_pop(self, buffer):
+        """Return the item at a randomized location `self._idx` in `buffer`."""
+        self.randomize(len(buffer))
+        ret, buffer[self._idx] = buffer[self._idx], buffer[-1]
+        buffer.pop()
+        return ret
+
+    def generate_item(self):
```

Comment:
I think we should keep the `generate_item` and `randomized_pop` functions as private, starting with "_", like:
https://github.com/Project-MONAI/MONAI/blob/dev/monai/data/iterable_dataset.py#L268
What do you think?

Thanks.

Reply:
Hi @Nic-Ma, these two APIs are generic and they'll be useful for cases where users don't want to use the hardcoded random state logic in `__iter__`.